torch/_inductor/lowering.py

```diff
@@ -6089,6 +6091,24 @@ def _make_reduction_inner(x, *, axis, keepdims, dtype, override_return_dtype):
             kept_idx.append(i)
             kept_sizes.append(size[i])
 
+    # For argmax/argmin compute logical indices when the tensor has non-contiguous layout.
+    should_compute_logical_index = False
+    if (
+        reduction_type in ("argmax", "argmin")
+        and len(reduced_sizes) > 1
+        and is_triton(x)
+    ):
+        if isinstance(x.data, PermuteView):
+            should_compute_logical_index = True
+        elif isinstance(x.data, ir.ReinterpretView) or (
+            isinstance(x.data, ir.StorageBox)
+            and isinstance(x.data.data, ir.InputBuffer)
```

Comment:
hmm, it works for graph input now but still does not work for internally generated buffers  (e.g. ComputedBuffer).

Here are a few choices:
1. either change `isinstance(x.data.data, ir.InputBuffer)` to `isinstance(x.data.data, ir.Buffer)`
2. or always use logical index disregarding the layout.  Logical index should still work even if the input for argmin/max is contiguous right?